app/controllers/election.ts

```diff
@@ -59,56 +90,69 @@ export const retrieveActive = (req, res) =>
       return res.status(200).json(election);
     });
 
-export const create = (req, res) =>
-  Election.create({
-    title: req.body.title,
-    description: req.body.description,
-    seats: req.body.seats,
-    type: req.body.type,
-    useStrict: req.body.useStrict,
-    physical: req.body.physical,
-  })
-    .then((election) => {
-      const alternatives = req.body.alternatives;
-      if (alternatives && alternatives.length) {
-        Promise.all(
-          alternatives.map((a) => {
-            a.election = election;
-            return Alternative.create(a);
-          })
-        ).then((createdAlternatives) => {
-          election.alternatives = createdAlternatives;
-          return election.save();
-        });
-      }
-      return election;
-    })
-    .then((election) => election.populate('alternatives').execPopulate())
-    .then((election) => res.status(201).json(election))
-    .catch((err) => {
-      if (err instanceof mongoose.Error.ValidationError) {
-        throw new errors.ValidationError(err.errors);
-      }
+export const create = async (
+  req: TypedRequestBody<CreateElection>,
```

Comment:
Ahh, yes that looks a lot better. But I think the reason for using this was that the generic from express is really shit to use. (especially since `{}` is not allowed to use due to eslint rules).